packages/compass-components/src/components/leafygreen.tsx

```diff
@@ -0,0 +1,178 @@
+import type React from 'react';
+
+import { withTheme } from '../hooks/use-theme';
+
+// This file exports `@leafygreen-ui` components and wraps some of
+// them with a listener to Compass' theme in the react context.
+
+// 1. Import the components we use from leafygreen.
+import { default as Badge } from '@leafygreen-ui/badge';
+import { default as Banner } from '@leafygreen-ui/banner';
+import { default as LeafyGreenButton } from '@leafygreen-ui/button';
+import { default as LeafyGreenCheckbox } from '@leafygreen-ui/checkbox';
+import { default as LeafyGreenCard } from '@leafygreen-ui/card';
+import { default as LeafyGreenConfirmationModal } from '@leafygreen-ui/confirmation-modal';
+import { default as Icon } from '@leafygreen-ui/icon';
+import { default as LeafyGreenIconButton } from '@leafygreen-ui/icon-button';
+import {
+  AtlasLogoMark,
+  MongoDBLogoMark,
+  MongoDBLogo,
+} from '@leafygreen-ui/logo';
+import { Menu, MenuSeparator, MenuItem } from '@leafygreen-ui/menu';
+import {
+  default as LeafyGreenModal,
+  Footer as LeafyGreenFooter,
+} from '@leafygreen-ui/modal';
+import { RadioBox, RadioBoxGroup } from '@leafygreen-ui/radio-box-group';
+import {
+  Radio,
+  RadioGroup as LeafyGreenRadioGroup,
+} from '@leafygreen-ui/radio-group';
+import {
+  SegmentedControl as LeafyGreenSegmentedControl,
+  SegmentedControlOption,
+} from '@leafygreen-ui/segmented-control';
+import {
+  Select as LeafyGreenSelect,
+  Option,
+  OptionGroup,
+} from '@leafygreen-ui/select';
+import {
+  Table as LeafyGreenTable,
+  TableHeader,
+  Row,
+  Cell,
+} from '@leafygreen-ui/table';
+import { Tabs as LeafyGreenTabs, Tab } from '@leafygreen-ui/tabs';
+import { default as LeafyGreenTextArea } from '@leafygreen-ui/text-area';
+import { default as LeafyGreenTextInput } from '@leafygreen-ui/text-input';
+import { default as Toast } from '@leafygreen-ui/toast';
+import { default as LeafyGreenToggle } from '@leafygreen-ui/toggle';
+import { default as LeafyGreenTooltip } from '@leafygreen-ui/tooltip';
+import {
+  H1,
+  H2,
+  H3,
+  Subtitle,
+  Body,
+  InlineCode,
+  InlineKeyCode,
+  Disclaimer,
+  Overline,
+  Label as LeafyGreenLabel,
+  Link,
+  Description as LeafyGreenDescription,
+} from '@leafygreen-ui/typography';
+
+// 2. Wrap the components that accept darkMode with Compass' theme.
+const Button = withTheme(
+  LeafyGreenButton as React.ComponentType<typeof LeafyGreenButton>
+) as typeof LeafyGreenButton;
```

Comment:
Ok, let me know when i can review again

Reply:
Added `withTheme` to the `FileInput` as an example/test of using the theme in a custom component, works smooth. I spent a while trying to get these types for these leafygreen components to work nicely yesterday and was having trouble finding a clean way without casting explicitly here. Do you think it's a blocker for merging? I'd be down to pair on looking at the `withTheme` hoc to see if there's anything we can do to avoid casts. If we do merge I'd definitely want to solidify these types before the epic is done, but for now we would be able to unblock theming other parts of this epic.

It appears typescript tries to compare the inferred react prop types with the typescript prop definition and that results in a typing/compile error. To get around this we explicitly cast them to their leafygreen types.

Here's examples of the two errors we were seeing:
```
Exported variable 'Toggle' has or is using name 'BaseToggleProps' from external module "/Users/rhys/Documents/mongodb/compass/node_modules/@leafygreen-ui/toggle/dist/Toggle" but cannot be named.
```

```
Argument of type 'typeof Toggle' is not assignable to parameter of type 'ComponentType<ToggleProps>'.
  Type 'typeof Toggle' is not assignable to type 'FunctionComponent<ToggleProps>'.
    Types of property 'propTypes' are incompatible.
      Type '{ size: Requireable<string>; darkMode: Requireable<boolean>; checked: Requireable<boolean>; disabled: Requireable<boolean>; className: Requireable<...>; onChange: Requireable<...>; onClick: Requireable<...>; }' is not assignable to type 'WeakValidationMap<ToggleProps> | undefined'.
        Type '{ size: Requireable<string>; darkMode: Requireable<boolean>; checked: Requireable<boolean>; disabled: Requireable<boolean>; className: Requireable<...>; onChange: Requireable<...>; onClick: Requireable<...>; }' is not assignable to type 'WeakValidationMap<Omit<BaseToggleProps & Omit<HTMLElementProps<"button", never>, keyof BaseToggleProps>, "aria-label" | "aria-labelledby"> & Required<...> & Partial<...>>'.
          The types of 'size[nominalTypeHack]' are incompatible between these types.
            Type '{ type: string | null | undefined; } | undefined' is not assignable to type '{ type: Size | null | undefined; } | undefined'.
              Type '{ type: string | null | undefined; }' is not assignable to type '{ type: Size | null | undefined; }'.
                Types of property 'type' are incompatible.
                  Type 'string | null | undefined' is not assignable to type 'Size | null | undefined'.
                    Type 'string' is not assignable to type 'Size | null | undefined'.
```

Reply:
@mcasimir Pushed an update to make the components that use the `withTheme` wrapper require an optional `darkMode` prop. Also added an explicit cast to the generic type in the hoc.
Re https://stackoverflow.com/a/58473012